plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -584,24 +616,62 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
-    public static LongWriteFunction trinoTimestampToOracleTimestampWriteFunction()
+    private static ObjectWriteFunction oracleLongTimestampWriteFunction(TimestampType timestampType)
+    {
+        int roundToPrecision = timestampType.getPrecision();
+        checkArgument(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
```

Comment:
nit: `MAX_SHORT_PRECISION` static import

Reply:
`MAX_SHORT_PRECISION ` is too general. I would recommend using `isShort` method.

Reply:
refactored